include/zephyr/net/net_pkt.h

```diff
@@ -1882,6 +1894,31 @@ int net_pkt_alloc_buffer(struct net_pkt *pkt,
 			 k_timeout_t timeout);
 #endif
 
+/**
+ * @brief Allocate buffer for a net_pkt and reserve some space in the first net_buf.
+ *
+ * @details: such allocator will take into account space necessary for headers,
+ *           MTU, and existing buffer (if any). Beware that, due to all these
+ *           criteria, the allocated size might be smaller/bigger than
+ *           requested one.
+ *
+ * @param pkt     The network packet requiring buffer to be allocated.
+ * @param size    The size of buffer being requested.
+ * @param reserve The L2 header size to reserve. This can be 0, in which case
+ *                the L2 header is placed into a separate net_buf.
+ * @param proto   The IP protocol type (can be 0 for none).
+ * @param timeout Maximum time to wait for an allocation.
+ *
+ * @return 0 on success, negative errno code otherwise.
+ */
+#if !defined(NET_PKT_DEBUG_ENABLED)
+int net_pkt_alloc_buffer_with_reserve(struct net_pkt *pkt,
```

Comment:
Afaik, you don't need a dedicated net_pkt function just to be able to call net_buf_reserve() because ... (see my comment in ethernet.c)